x/blobstream/types/validator.go

```diff
@@ -129,13 +131,35 @@ func (ibv InternalBridgeValidators) PowerDiff(c InternalBridgeValidators) float6
 		}
 	}
 
-	var delta float64
+	var delta int64
 	for _, v := range powers {
 		// NOTE: we care about the absolute value of the changes
-		delta += math.Abs(float64(v))
+		if v < 0 {
+			v = -v
+		}
+		delta += v
 	}
 
-	return math.Abs(delta / float64(math.MaxUint32))
+	decDelta := sdk.NewDec(int64(delta))
+	decMaxUint32 := sdk.NewDec(math.MaxUint32)
+	q := decDelta.Quo(decMaxUint32)
+
+	return q.Abs()
+}
+
+// Int64Abs returns the absolute value of an int64.
+// For math.MinInt64, it returns math.MaxInt64.
+// This is because the absolute value of math.inInt64 is
+// 1 greater than math.MaxInt64. So, we return math.MaxInt64 as the closest
+// approximation which has 1 bit of error.
+func Int64Abs(v int64) int64 {
+	if v < 0 {
+		if v == math.MinInt64 { // avoid overflow
+			return -(v + 1) // i.e., math.MaxInt64
```

Comment:
[Note to reviwers @evan-forbes @SweeXordious ] This is a tricky and problematic corner case, without this line, the negative of `-9223372036854775808` (math,MinInt64) equals itself but not `9223372036854775808`. The reason is explained in the function description.


Upon further thoughts, another issue came to my mind: if the sum of the absolute power diffs exceeds the maximum value representable by `int64` then there will be overflow issues (do we expect this happen?). This is while using `float64` it is less likely since the largest value representable by `float64` is much larger than the max value that an `int64` can accomodate. So, we may want to reconsider the original version in which delta was `float64` and we have been using math.Abs() instead. Please let me know your thoughts.

Reply:
Please note that float64 calculations also introduce their own errors, as illustrated below:

```go
v1 := math.MinInt64 // -9223372036854775808
v2 := float64(math.MinInt64) // -9223372036854776000
```

When examining the values of `v1` and `v2`, it becomes evident that the `float64` representation of `v1` does not precisely match its `int64` value.

Reply:
I don't think we have anything to worry about as long as we use normal values (most testnets have 1*10^15 total utia and 1*10^9 voting power), this is just being extra careful. 

I'm definitely okay with shipping and testing smaller changes, and was incorrect to assume that staying with int64 was a straightforward switch, will defer to the author on that

Reply:
I'm tempted to just use big.Int for everything

Reply:
Hum, yes `big.Int` or we just use `sdk.Dec` wrapper for everything, even the map of powers created in `powers := map[string]int64{}`. That would help us avoid all of these conversions but not sure about the consequences of it in functions calling the `PowerDiff`.

Reply:
In the setting you mentioned i.e.,
> (most testnets have 1x10^15 total utia and 1x10^9 voting power)

it should be fine to use `int64` to hold the sum of power diffs, but if in Arabica it can get larger and the upper bound is not known yet, then I'd vote for keeping delta as `float64` just to avoid potential overflows.

@SweeXordious Once you mentioned powers are normalized? to what range they are normalized to? What is the max power value after normalization? Basically, what is the max power value that `PowerDiff` function can receive as its input? Is the sum of powers capped?

> I'm tempted to just use big.Int for everything

This or `sdk.Dec` which underneath is a `big.Int`.

Reply:
> @SweeXordious Once you mentioned powers are normalized, to what range they are normalized to? What is the max power value after normalization? Basically, what is the max power value that PowerDiff function can receive as its input? Is the sum of powers capped?

They're normalized to `2^32`:
https://github.com/celestiaorg/celestia-app/blob/fc348bf395e0c56787aa0fe587275adf2f04c40b/x/blobstream/keeper/keeper_valset.go#L136-L161

Reply:
Thank you for providing the context. While I believe that using `int64` with this normalization would have also been functional, I have since successfully refactored the `PowerDiff` function to utilize `sdk.Dec` from the outset (managed to resolve the bug I mentioned earlier). Regarding the code review, everything seems to be in order, except for the potential overflow issue when casting from `uint64` to `int64`; we have partially discussed it, however, it is out of scope of this PR and merits its own separate PR.